onnx/checker.cc

```diff
@@ -360,6 +360,98 @@ void check_graph(
   }
 }
 
+Status check_function(
```

Comment:
All other checker functions are using exceptions, this new `check_function` should be consistent with the others.

Reply:
I explained below. a while ago, I talked to Dima somehow to introduce "Status" class into onnx for error handling instead of throwing exceptions. Now, I'm adding it, it will take some time to change existing exception usage though.

Reply:
btw, I can move this function into function.h/cc for now, if you feel that's better. :)

Reply:
Oh I didn't know you had discussions with @dzhulgakov already. My main concern is having **inconsistent** styles of error handling in our api. What was the problem of using exceptions?

Reply:
Whether using exception for error handling, I believe it could be a long-debate question. As a core ONNX lib, I personally think it's better to not throw exceptions, especially in some of our static registration codes, throwing exceptions or "abort" process is not a good idea. I agree with you that we should have consistent error handling. I'm targeting to move all error handling to use "Status", instead of throwing exceptions. Thoughts? Again, if there's different opinion here, to avoid long debate on this PR, I can move this function into function.h instead of putting it here. Which way do you prefer please?

Reply:
I don't think moving into a separate file will help the issue of inconsistency.
If we are doing this at the beginning, I probably won't be against using status code (as you said it's a long-debate question that I don't want us get into it if possible). However, considering this api has been established for a while (so there could be many users already), I prefer to not change it.

Reply:
We can't allow "exception" and "abort" in our static registration code, in windows ml scenario. That's why we should avoid using exception as much as possible. To me, it's acheiveable to not use exception in onnx. Make sense?